File: Backend/src/main/java/com/hms/elite_haven/service/RoomPhotoService.java
package com.hms.elite_haven.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.hms.elite_haven.dao.RoomPhotoDao;
import com.hms.elite_haven.dao.entity.RoomPhotoEntity;

@Service
public class RoomPhotoService {

    @Autowired
    private RoomPhotoDao roomPhotoDao;

    // Add a new photo (input is the entity)
    public RoomPhotoEntity addPhoto(RoomPhotoEntity roomPhoto) {
        return roomPhotoDao.save(roomPhoto);
    }

    // Get all photos for a room
    public List<RoomPhotoEntity> getPhotosByRoomId(Long roomId) {
        return roomPhotoDao.findByRoom_RoomId(roomId);
    }

    // Get a specific photo by ID
    public Optional<RoomPhotoEntity> getPhotoById(Long photoId) {
        return roomPhotoDao.findById(photoId);
    }

    // Update photo details (like marking it as primary)
    public RoomPhotoEntity updatePhoto(RoomPhotoEntity roomPhoto) {
        return roomPhotoDao.save(roomPhoto);
    }

    // Delete photo by ID
    public void deletePhoto(Long photoId) {
        roomPhotoDao.deleteById(photoId);
    }
}

File: Frontend/src/app/services/room-photo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RoomPhotoModel } from '../models/room.model';

@Injectable({
  providedIn: 'root',
})
export class RoomPhotoService {
  private baseUrl = '/api/v1';

  constructor(private http: HttpClient) {}

  addPhoto(roomPhoto: RoomPhotoModel): Observable<RoomPhotoModel> {
    return this.http.post<RoomPhotoModel>(
      `${this.baseUrl}/add_photo`,
      roomPhoto
    );
  }

  getPhotosByRoomId(roomId: number): Observable<RoomPhotoModel[]> {
    return this.http.get<RoomPhotoModel[]>(
      `${this.baseUrl}/room_photos/${roomId}`
    );
  }

  getPhotoById(photoId: number): Observable<RoomPhotoModel> {
    return this.http.get<RoomPhotoModel>(`${this.baseUrl}/photo/${photoId}`);
  }

  updatePhoto(roomPhoto: RoomPhotoModel): Observable<RoomPhotoModel> {
    return this.http.put<RoomPhotoModel>(
      `${this.baseUrl}/update_photo`,
      roomPhoto
    );
  }

  deletePhoto(photoId: number): Observable<string> {
    return this.http.delete<string>(`${this.baseUrl}/delete_photo/${photoId}`);
  }
}
